### ISSUE TITLE
Apostrophe's break how Goals amounts are output

### ISSUE THREAD

**Issue by mathetos** [WordImpress/Give#914]
Related to #913 when an apostrophe is used as a thousands separator, the Goal amount gets ruined. With DEBUG on, I get this error:

```
Notice: A non well formed numeric value encountered in 
...\wp-content\plugins\give\includes\formatting.php on line 169
```

Additionally, while my goal is set at $10,000 (but with apostrophe as the separator), here's what it looks like:
![core-thousands-separator-apostrophe-goal-amount-bug](https://cloud.githubusercontent.com/assets/930724/17757975/4c748f40-64a0-11e6-9d48-5770822fbb2e.png)

And that's $47.00 raised so far, but it makes it look like $47 out of $10 total raised.

**Comment by DevinWalker** [comment]
@ravinderk I'm still seeing this issue after merging #928:

![screen shot 2016-08-22 at 3 48 41 pm](https://cloud.githubusercontent.com/assets/1571635/17874472/ac4b77f6-6880-11e6-9f38-ddef73bb29b1.png)

![screen shot 2016-08-22 at 3 52 53 pm](https://cloud.githubusercontent.com/assets/1571635/17874476/b1ed6462-6880-11e6-919a-54ab50be8d08.png)

**Comment by DevinWalker** [comment]
Fixed with #942
